src/kademlia/dht_tracker.cpp

```diff
@@ -285,7 +370,11 @@ namespace libtorrent { namespace dht
 	void dht_tracker::get_item(sha1_hash const& target
 		, boost::function<void(item const&)> cb)
 	{
-		m_dht.get_item(target, boost::bind(&get_immutable_item_callback, _1, cb));
+		boost::shared_ptr<bool> item_posted = boost::make_shared<bool>(false);
```

Comment:
There's no existing object with the appropriate lifetime. I could create a pool allocator for them in the dht_tracker, but to me it seems like premature optimization since no existing use case for BEP44 is going to be generating large numbers of get/put requests.